src/ArcanaPrime.sol

```diff
@@ -225,13 +241,13 @@ contract ArcanaPrime is ERC721AQueryable, ERC721ABurnable, Ownable, OperatorFilt
     /// @notice Mints part of the supply in the community wallet that Arcana owns. Note: Likely hidden from OpenSea due to Aux.
     /// @dev Only the Owner of the smart contract can call this function
     /// @param _communityWalletPublicKey The address of the community wallet
-    function mintWarChestReserve(address _communityWalletPublicKey)
+    function mintWarChestReserve(address _communityWalletPublicKey, uint256 _supply)
         external
         payable
-        isBelowMaxSupply(WAR_CHEST_SUPPLY)
+        isBelowMaxSupply(_supply)
```

Comment:
Centralisation risk: a malicious or a compromised owner can now mint as much NFTs (up to `MAX_SUPPLY`) whenever he wants to. If an NFT mint is going well, let's say 9000 out of 10000 NFTs were minted very quickly, the owner can mint the other 1000 for him and then dump them on the market for profit

Reply:
added a public nextUnlockTs that prevents us from minting treasury at any time which handicaps the above exploit mentioned.